File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n=len(nums)
        map={}
        for i in range(n):
            map[nums[i]]=i
        for i in range(n):
            pair=target-nums[i]
            if pair in map and map[pair]!=i:
                return [i,map[pair]]
        return[]
        

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int>map;
        for(int i=0;i<nums.size();i++)
        {
            map[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++)
        {
            int pair=target-nums[i];
            if(map[pair] && map[pair]!=i)
            {
                return{i,map[pair]};
            }
        }
        return{};
        
    }
};